extensions/ql-vscode/src/view/common/Alert.tsx

```diff
@@ -0,0 +1,98 @@
+import * as React from 'react';
+import { ReactNode } from 'react';
+import styled from 'styled-components';
+
+type ContainerProps = {
+  type: 'warning' | 'error';
+  inverse?: boolean;
+};
+
+const getBackgroundColor = ({ type, inverse }: ContainerProps): string => {
+  if (!inverse) {
+    return 'var(--vscode-notifications-background)';
+  }
+
+  switch (type) {
+    case 'warning':
+      return 'var(--vscode-editorWarning-foreground)';
+    case 'error':
+      return 'var(--vscode-debugExceptionWidget-border)';
+  }
+};
+
+const getTextColor = ({ type, inverse }: ContainerProps): string => {
+  if (!inverse) {
+    return 'var(--vscode-tab-activeForeground)';
```

Comment:
```suggestion
    return 'var(--vscode-editor-foreground)';
```
Changes from #ffffff to #d4d4d4

Probably the biggest change in these suggestions. Does VSCode use pure white for text in dark mode? From what I can tell it does seem to be slightly darker generally, which is nicer on the eyes.

Reply:
VSCode does use light gray instead of white and it makes sense to be consistent there, so I'll change this.